isobus/src/can_message.cpp

```diff
@@ -39,11 +39,36 @@ namespace isobus
 		return source;
 	}
 
+	bool CANMessage::has_valid_source_control_function() const
+	{
+		return nullptr != source;
```

Comment:
"Valid" might be inaccurate, it's just not `nullptr`... might need to also check the CF has claimed since the last address claiming session with the appropriate function?